inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/service/cache/AuditRouteCache.java

```diff
@@ -0,0 +1,179 @@
+            LOGGER.error("Failed to create DataSource in AuditRouteCache init", e);
+        }
+
+        timerExecutor.scheduleWithFixedDelay(() -> {
+            try {
+                updateAuditRouteCache(querySQL);
+            } catch (Exception e) {
+                LOGGER.error("Exception occurred during audit route cache update", e);
+            }
+        }, 0, PERIOD_MS, TimeUnit.MILLISECONDS);
+    }
+
+    private AuditDataSource createDataSource() {
+        JdbcConfig jdbcConfig = JdbcUtils.buildMysqlConfig();
+
+        BasicDataSource dataSource = new BasicDataSource();
+        dataSource.setDriverClassName(jdbcConfig.getDriverClass());
+        dataSource.setUrl(jdbcConfig.getJdbcUrl());
+        dataSource.setUsername(jdbcConfig.getUserName());
+        dataSource.setPassword(jdbcConfig.getPassword());
+        dataSource.setValidationQuery("SELECT 1");
+
+        return new AuditDataSource(jdbcConfig.getJdbcUrl(), dataSource);
+    }
+
+    private void updateAuditRouteCache(String querySQL) {
+        Map<String, List<AuditRoute>> auditRouteList = new HashMap<>();
+
+        try (Connection connection = auditDataSource.getDataSource().getConnection();
+                Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(querySQL)) {
+
+            while (resultSet.next()) {
+                String address = resultSet.getString(ADDRESS);
+                if (address == null || address.trim().isEmpty()) {
+                    continue;
+                }
+
+                String auditId = StringUtils.trimToNull(resultSet.getString(AUDIT_ID_INCLUDE));
+                String includeGroupId = StringUtils.trimToNull(resultSet.getString(GROUP_ID_INCLUDE));
+                String excludeGroupId = StringUtils.trimToNull(resultSet.getString(GROUP_ID_EXCLUDE));
+
+                if (!isValidRegexOrLog(auditId)
+                        || !isValidRegexOrLog(includeGroupId)
+                        || !isValidRegexOrLog(excludeGroupId)) {
+                    LOGGER.error(
+                            "Skipping invalid regex entry: address={}, auditId={}, includeGroupId={}, excludeGroupId={}",
+                            address, auditId, includeGroupId, excludeGroupId);
+                    continue;
+                }
+
+                AuditRoute data = new AuditRoute();
+                data.setAddress(address);
+                data.setAuditId(auditId);
+                data.setInlongGroupIdsInclude(includeGroupId);
+                data.setInlongGroupIdsExclude(excludeGroupId);
+
+                auditRouteList.computeIfAbsent(address, key -> new ArrayList<>()).add(data);
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Failed to update audit route cache", e);
+            return;
+        }
+
+        if (!auditRouteList.isEmpty()) {
+            auditRouteCache = new ConcurrentHashMap<>(auditRouteList);
```

Comment:
Map<String, List<AuditRoute>> auditRouteList = new HashMap<>();
            auditRouteCache = new ConcurrentHashMap<>(auditRouteList);

It is better.
        Map<String, List<AuditRoute>> auditRouteList = new ConcurrentHashMap<>();
            auditRouteCache = auditRouteList;